packages/yourloops/components/menus/hcp-team-scope-menu.tsx

```diff
@@ -0,0 +1,217 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { type FunctionComponent, useState } from 'react'
+import Box from '@mui/material/Box'
+import { makeStyles } from 'tss-react/mui'
+import { type Theme, useTheme } from '@mui/material/styles'
+import { type Team, useTeam } from '../../lib/team'
+import PersonIcon from '@mui/icons-material/Person'
+import { useSelectedTeamContext } from '../../lib/selected-team/selected-team.provider'
+import { TeamType } from '../../lib/team/models/enums/team-type.enum'
+import { useTranslation } from 'react-i18next'
+import { type TeamEditModalContentProps } from '../../pages/hcp/types'
+import { useAlert } from '../utils/snackbar'
+import GroupsOutlinedIcon from '@mui/icons-material/GroupsOutlined'
+import { useNavigate } from 'react-router-dom'
+import TeamUtils from '../../lib/team/team.util'
+import useMediaQuery from '@mui/material/useMediaQuery'
+import IconButton from '@mui/material/IconButton'
+import Button from '@mui/material/Button'
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown'
+import Typography from '@mui/material/Typography'
+import MenuLayout from '../../layout/menu-layout'
+import MenuItem from '@mui/material/MenuItem'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import ListItemText from '@mui/material/ListItemText'
+import Divider from '@mui/material/Divider'
+import AddIcon from '@mui/icons-material/Add'
+import TeamEditDialog from '../../pages/hcp/team-edit-dialog'
+
+const classes = makeStyles()((theme: Theme) => ({
+  sectionTitle: {
+    fontWeight: 'bold',
+    paddingLeft: theme.spacing(2),
+    paddingBottom: theme.spacing(1)
+  },
+  menu: {
+    paddingTop: theme.spacing(1),
+    paddingBottom: theme.spacing(1)
+  }
+}))
+
+export const HcpTeamScopeMenu: FunctionComponent = () => {
+  const { t } = useTranslation('yourloops')
+  const {
+    classes: {
+      sectionTitle,
+      menu
+    }
+  } = classes()
+  const { getMedicalTeams, getPrivateTeam } = useTeam()
+  const { selectTeam, selectedTeamId } = useSelectedTeamContext()
+  const { createTeam } = useTeam()
+  const alert = useAlert()
+  const navigate = useNavigate()
+  const [teamCreationDialogData, setTeamCreationDialogData] = React.useState<TeamEditModalContentProps | null>(null)
+  const theme = useTheme()
+  const isMobile: boolean = useMediaQuery(theme.breakpoints.only('xs'))
+
+  const privateTeam = getPrivateTeam()
+  const sortedMedicalTeams = TeamUtils.sortTeamsByName(getMedicalTeams())
+  const hasMedicalTeams = sortedMedicalTeams.length > 0
+  const availableTeams = [privateTeam, ...sortedMedicalTeams]
+
+  const selectedTeam = availableTeams.find((team: Team) => team.id === selectedTeamId)
```

Comment:
I'm not sure this case can really happen because the value is initialised with a value (see the method `getDefaultTeamId()` in `selected-team.hook.tsx`). What do you think, am I missing something?

Reply:
I don't think that it can happen. But I think that it is possible in future development to accidentally "break" this. I let you decide if you think that it is worth it or not to be prepared for such a case